pyg_lib/csrc/random/cpu/randint_engine.h

```diff
@@ -0,0 +1,55 @@
+#include <torch/torch.h>
+
+#include <limits.h>
+
+namespace pyg {
+namespace random {
+const int RAND_PREFETCH_THRESHOLD = 128;
+const int RAND_PREFETCH_BITS = 64;
+template <typename T>
+class RandintEngine {
+ public:
+  RandintEngine() : size_(RAND_PREFETCH_THRESHOLD), bits_(64) {
+    prefetch_randint_ = torch::randint(
+        0L, std::numeric_limits<int64_t>::max(), {RAND_PREFETCH_THRESHOLD},
+        torch::TensorOptions().dtype(torch::kInt64));
+  }
+  template <unsigned B>
+  T rand(T range) {
+    T num;
+    if (bits_ < B) {
+      if (size_ > 0) {
+        size_--;
+        bits_ = 64;
+      } else {
+        prefetch_randint_ = torch::randint(
+            0L, std::numeric_limits<int64_t>::max(), {RAND_PREFETCH_THRESHOLD},
+            torch::TensorOptions().dtype(torch::kInt64));
+        size_ = RAND_PREFETCH_THRESHOLD;
+        bits_ = RAND_PREFETCH_BITS;
+      }
+    }
+    int64_t* prefetch_ptr = prefetch_randint_.data_ptr<int64_t>();
```

Comment:
we can define this as a class variable and reset it whenever we reset `prefetch_randint_`.